### PR TITLE
Added support for Match function for Semi-Structured data.

### PR DESCRIPTION
**Pull request by pragyasri-gs**
#### What type of PR is this?
Improvement PR

#### What does this PR do ?
Added support for Match function for Semi Structured Data.
Following changes have been implemented :
1. Added pure to SQL conversion for match queries.
2. Added relevant testcases for Snowflake,H2 and MemSQL database.

#### Does this PR introduce a user-facing change?
Yes

### CODE REVIEW

File: legend-engine-xts-relationalStore/legend-engine-xt-relationalStore-dbExtension/legend-engine-xt-relationalStore-snowflake/legend-engine-xt-relationalStore-snowflake-execution/src/test/java/org/finos/legend/engine/plan/execution/stores/relational/test/semiStructured/TestSnowflakeSemiStructuredMatching.java [hunk 1]
```diff
@@ -0,0 +1,177 @@
+// Copyright 2022 Goldman Sachs
```
[the remark this answers is not in the document]
Comment (pragyasri-gs) [reply legend-engine-xts-relationalStore/legend-engine-xt-relationalStore-dbExtension/legend-engine-xt-relationalStore-snowflake/legend-engine-xt-relationalStore-snowflake-execution/src/test/java/org/finos/legend/engine/plan/execution/stores/relational/test/semiStructured/TestSnowflakeSemiStructuredMatching.java]: Already done.

File: legend-engine-xts-relationalStore/legend-engine-xt-relationalStore-execution/legend-engine-xt-relationalStore-executionPlan/src/test/java/org/finos/legend/engine/plan/execution/stores/relational/test/semiStructured/TestSemiStructuredMatching.java [hunk 2]
```diff
@@ -0,0 +1,219 @@
+// Copyright 2022 Goldman Sachs
```
[the remark this answers is not in the document]
Comment (pragyasri-gs) [reply legend-engine-xts-relationalStore/legend-engine-xt-relationalStore-execution/legend-engine-xt-relationalStore-executionPlan/src/test/java/org/finos/legend/engine/plan/execution/stores/relational/test/semiStructured/TestSemiStructuredMatching.java]: Already done.

File: legend-engine-xts-relationalStore/legend-engine-xt-relationalStore-execution/legend-engine-xt-relationalStore-executionPlan/src/test/resources/org/finos/legend/engine/plan/execution/stores/relational/test/semiStructured/semiStructuredMatching.pure [hunk 3]
```diff
@@ -0,0 +1,349 @@
+// Copyright 2022 Goldman Sachs
```
[the remark this answers is not in the document]
Comment (pragyasri-gs) [reply legend-engine-xts-relationalStore/legend-engine-xt-relationalStore-execution/legend-engine-xt-relationalStore-executionPlan/src/test/resources/org/finos/legend/engine/plan/execution/stores/relational/test/semiStructured/semiStructuredMatching.pure]: Already done.

**Review by gs-ssh16 — APPROVED** [review]

**Pull request merged**